src/main/java/org/wildfly/swarm/proc/Monitor.java

```diff
@@ -392,7 +392,9 @@ private void measureMemory(String id, String uid, Collector collector) throws Ex
         final ProcessFinder processFinder = new ProcessFinder(sigar);
         long pid = processFinder.findSingleProcess("State.Name.eq=java,Args.1.ct="+uid);
 
-        Jcmd.gc(pid);
+        for (int i = 0; i < 10; i++) {
+            Jcmd.gc(pid);
+        }
```

Comment:
Interesting...

Reply:
:-)

Reply:
more helps better

Reply:
My very non-scientific experience is that 1 GC cycle isn't enough to collect everything. I'm not sure if we have many objects with finalizers (or in reference queues), but that's one reason why at least 2 GC cycles are required to clean up properly.